source/extensions/quic/crypto_stream/envoy_quic_crypto_server_stream.cc

```diff
@@ -8,10 +11,12 @@ EnvoyQuicCryptoServerStreamFactoryImpl::createEnvoyQuicCryptoServerStream(
     const quic::QuicCryptoServerConfig* crypto_config,
     quic::QuicCompressedCertsCache* compressed_certs_cache, quic::QuicSession* session,
     quic::QuicCryptoServerStreamBase::Helper* helper,
-    // Though this extension doesn't use the two parameters below, they might be used by
-    // downstreams. Do not remove them.
-    OptRef<const Network::DownstreamTransportSocketFactory> /*transport_socket_factory*/,
-    Envoy::Event::Dispatcher& /*dispatcher*/) {
+    OptRef<const Network::DownstreamTransportSocketFactory> transport_socket_factory,
+    Envoy::Event::Dispatcher& dispatcher) {
+  if (session->connection()->version().handshake_protocol == quic::PROTOCOL_TLS1_3) {
```

Comment:
This is always true as Envoy only support RFC and draft29 implementations.

Reply:
got it, thanks! I confused when I found some of tests provide other implementation https://github.com/envoyproxy/envoy/blob/372a262894f028431cb84f3a1d361fb8fb6a2518/test/common/quic/envoy_quic_server_session_test.cc#L128-L137

But at same time, I didn't we have any integration test for quic crypto. 

That is probably something we can cleanup